src/protobuf-net/Meta/MetaType.cs

```diff
@@ -2025,19 +2024,22 @@ public bool IsGroup
 
                 bool haveWrittenZero = false;
                 // write zero values **first**
+
+                string enumPrefix = flags.HasFlag(SchemaGenerationFlags.PrefixEnumValuesWithEnumName) ? GetSchemaTypeName(callstack) + "_" : "";
                 foreach (var member in enums)
                 {
                     var parsed = member.TryGetInt32();
                     if (parsed.HasValue && parsed.Value == 0)
                     {
-                        NewLine(builder, indent + 1).Append(member.Name).Append(" = 0;");
+                        NewLine(builder, indent + 1).Append(enumPrefix + member.Name).Append(" = 0;");

```

Comment:
Can this be two separate append calls, to prevent the extra alloc? Not really going to make a huge difference in real terms, but since we have the builder...